BootloaderCorePkg/BootloaderCorePkg.dsc

```diff
@@ -343,8 +343,14 @@
       BoardInitLib | Platform/$(BOARD_PKG_NAME)/Library/Stage2BoardInitLib/Stage2BoardInitLib.inf
   }
 
+  BootloaderCommonPkg/Library/GraphicsLib/GraphicsLib.inf {
+    <LibraryClasses>
+      BaseMemoryLib| MdePkg/Library/BaseMemoryLibSse2/BaseMemoryLibSse2.inf
+  }
+
   PayloadPkg/OsLoader/OsLoader.inf {
     <LibraryClasses>
+      BaseMemoryLib       | MdePkg/Library/BaseMemoryLibSse2/BaseMemoryLibSse2.inf

```

Comment:
Do we have perf benefit in OsLoader as well? How about Stage2?

Reply:
In OS Loader I see it saving time in the "Parse partition info" section (hundreds of milliseconds but unfortunately this is using debug build on APL, having issue with release build boot). Stage2 is having some issue in PCI enumeration if I use it for all of Stage2 code (but when overriding for the PciEnumerationLib specifically and not all of Stage2 it works so the issue is somewhere else in Stage2). Will try APL release build boot again to get some numbers.

Reply:
Needed to use an IAS image to boot (forgot that traditional Linux is debug only). Will post some numbers soon.

Reply:
BTW, I just looked into PCI enumeration issue with SSE2 memory lib.   It turns out it is using the lib to copy from MMIO to normal memory using SSE2 instruction.   PCI CFG MMIO needs to be accessed as DWORD.  
By changing the code as below,  it can work fine .  I think we can consider it and switch to use SSE2 lib for full Stage2.
PciDevicePresent ()
{
   // CopyMem (...)
    Src = (UINT32 *) ((UINTN)PcdGet64 (PcdPciExpressBaseAddress) + PCI_EXPRESS_LIB_ADDRESS (Bus, Device, Func, 0));
    Dst = (UINT32 *) Pci;
    for (Loop = 0; Loop < sizeof(PCI_TYPE00); Loop += 4) {
      *Dst++ = *Src++;
    }                         PCI_EXPRESS_LIB_ADDRESS (Bus, Device, Func, 0)), sizeof (PCI_TYPE00));
    return EFI_SUCCESS;
  }
}

Reply:
Just tested the patch, it is working, will add the change to the review, thanks Maurice!

For DSC file change, do you prefer to override Stage2 & OS Loader or update BaseMemoryLib to SSE2 library and override Stage1A & Stage1B to use non-SSE2 library?

Reply:
I think it might be better to use SSE2 lib by default for performance reason.
And then only override it for Stage1A/Stage1B.